tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorService.cs

```diff
@@ -321,8 +330,12 @@ public virtual void RecordHistory(CommandAst history)
         /// Requests a list of popular commands from service. These commands are used as fall back suggestion
         /// if none of the predictions fit for the current input. This method should be called once per session.
         /// </summary>
-        protected virtual void RequestAllPredictiveCommands()
+        protected virtual void RequestAllPredictiveCommands(Version version)
         {
+            if(version == null)
+            {
+                version = DefaultModelVersion;
```

Comment:
Do you want to make `_commandsEndpoint` below to interpolate with `version`?

Reply:
Ah, I see what you are saying. Actually upon further review, there is no need for this parameter for the method. We have access to the `_azContext` value where this parameter is coming from anyway. I will remove these parameters and use the `_azContext` object instead. This way, there is no confusion on what version is used in the `_commandsEndpoint` since it is set once from the constructor based on the `_azContext` object.